camelot/core/item_model/list_proxy.py

```diff
@@ -139,7 +139,7 @@ def index(self, obj):
                 # If the object's index, despite the object itself not being in the indexed objects, is present in the indexed_objects,
                 # this might indicate that another old object was removed from _objects outside the proxy's interface, which did not remove it from the indexed objects.
                 # So in this case we reassign the new object a new index at the end:
-                i = self._length
+                i = len(self._indexed_objects) // 2
```

Comment:
Dit is vermoed ik de enige correct manier, aangezien _length None kan zijn.

Reply:
ja, ik denk het ook

enige case waar ik nog aan denk (maar is eigenlijk nt relevant), is dat vanaf dit punt, eigenlijk elke extensie vd indexed objects moet rekening houden met het feit dat er een object is 'tussengezet'.

suppose dat het nieuwe object nt aan het einde was toegevoegd, maar bvb in het midden, terwijl de hele lijst nog nt was geindexeerd.

maar di eerder een theoretische bedenking